boards: add sparkfun_micromod_stm32

alvarowolfx opened the pull request:
Adds board definitions for Sparkfun STM32 MicroMod.

Tested with the following samples:
* basic/blinky
* subsys/usb/console
* sensor/bme680

Signed-off-by: Alvaro Viebrantz <alvarowolfx@gmail.com>

erwango commented on boards/arm/sparkfun_micromod_stm32/sparkfun_micromod_stm32.dts:
```diff
@@ -0,0 +1,93 @@
+	status = "okay";
+};
+
+&rtc {
+	status = "okay";
+};
```
I'm not seeing any mention to RTC otherwise in the board porting. Is that intended to enable it?

erwango replied:
You can test it using counter api test /sample

alvarowolfx replied:
Tested here the `samples/drivers/counter/alarm` and it worked.

`west build -b sparkfun_micromod_stm32 samples/drivers/counter/alarm -p`

```
*** Booting Zephyr OS build zephyr-v3.0.0-3807-g6ec92aa5232a  ***
Counter alarm sample

Set alarm in 2 sec (2 ticks)
!!! Alarm !!!
Now: 3
Set alarm in 4 sec (4 ticks)
!!! Alarm !!!
Now: 8
Set alarm in 8 sec (8 ticks)
```

erwango commented:
@alvarowolfx Thanks for this contribution. Looks fine to be but please add a commit message (this will fix the compliance issue reported)

alvarowolfx committed:
boards: add sparkfun_micromod_stm32

Adds board definitions for Sparkfun STM32 MicroMod.

Tested with the following samples:

basic/blinky
subsys/usb/console
sensor/bme680

Signed-off-by: Alvaro Viebrantz <alvarowolfx@gmail.com>

MaureenHelm commented on boards/arm/sparkfun_micromod_stm32/doc/index.rst:
```diff
@@ -0,0 +1,123 @@
+Overview
+********
+
+The SparkFun MicroMod STM32 Processor Board is ready to rock your MicroMod world with its ARM® Cortex®-M4 32-bit RISC core! This little Processor Board provides you with an economical and easy to use development platform if you're needing more power with minimal working space. With the M.2 MicroMod connector, connecting your STM32 Processor is a breeze. Simply match up the key on your processor's ...
+
+As we previously stated; the STM32F405 Processor is based on the high-performance ARM® Cortex®-M4 32-bit RISC core and can operate at a frequency of up to 168 MHz. This core features a floating point unit (FPU) single precision which supports all ARM single precision data-processing instructions and data types. It also implements a full set of DSP instructions and a memory protection unit (MPU) w ...
```
Please shorten the long lines to <100 characters

MaureenHelm commented on boards/arm/sparkfun_micromod_stm32/doc/index.rst:
```diff
@@ -0,0 +1,123 @@
+      :goals: build
+      :compact:
+
+#. On any of the MicroMod Baseboards, hold the boot button, press reset and release the boot button to put the device in bootloader mode.
```
Please shorten the lone line

erwango commented:
@alvarowolfx Do you intend to continue this PR ?

mmahadevan108 commented:
@alvarowolfx could you kindly take a look at the comments and help move this PR ahead.
I have cleared the 3.2 milestone as we are past the feature freeze date, please pick a new milestone for a future release.
